src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```
@@ -1028,6 +1029,134 @@ private TwoWindingsTransformer createTwoWindingsTransformer(Network network, Vol
+                            + "_" + cnxRank++)
+                        .setNode1(rank1)
+                        .setNode2(preBreakerRank)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+
+                    voltageLevel.getNodeBreakerView().newBreaker()
+                        .setKind(switchKind)
+                        .setId("breaker_" + voltageLevelCreationInfos.getEquipmentId() + "_" + cnxRank++)
+                        .setNode1(preBreakerRank)
+                        .setNode2(postBreakerRank)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+
+                    voltageLevel.getNodeBreakerView().newDisconnector()
+                        .setKind(SwitchKind.DISCONNECTOR)
+                        .setId("switch_" + voltageLevelCreationInfos.getEquipmentId() + "_" + toBBSId + "_" + fromBBSId
+                            + "_" + cnxRank++)
+                        .setNode1(postBreakerRank)
+                        .setNode2(rank2)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+                } else {
+                    voltageLevel.getNodeBreakerView().newDisconnector()
+                        .setKind(switchKind)
+                        .setId("switch_" + voltageLevelCreationInfos.getEquipmentId() + "_" + fromBBSId + "_" + toBBSId
+                            + "_" + cnxRank++)
+                        .setNode1(rank1)
+                        .setNode2(rank2)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+                }
```

Maybe throw an exception if SwitchKind is not BREAKER or DISCONNECTOR ?

---

done